spython/main/parse/parsers/docker.py

```diff
@@ -9,25 +9,57 @@
 import os
 import re
 
-from .environment import parse_env
-from .recipe import Recipe
 from spython.logger import bot
+from .base import ParserBase
 
-class DockerRecipe(Recipe):
 
-    def __init__(self, recipe=None):
-        '''a Docker recipe parses a Docker Recipe into the expected fields of
-           labels, environment, and install/runtime commands. We save working
-           directory as we parse, and the last one can be added to the runscript
-           of a Singularity recipe.
+class DockerParser(ParserBase):
+
+    name = 'docker'
+
+    def __init__(self, recipe='Dockerfile', load=True):
```

Comment:
Argument name: `recipe` is used for internal `Recipe`. Should rather be `filename`

Reply:
Agreed.